internal/database/mysql.go

```diff
@@ -138,16 +141,23 @@ func (db *MySQLDatabase) getOpenID() int {
 }
 
 func (db *MySQLDatabase) AddUser(username string, email string, password string) error {
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
```

Comment:
Could you explain where the salting occurs?

Reply:
This function will generate the salt.